### DISCUSSION TITLE
[Technical Questions] Pull request failing REUSE test

### DISCUSSION THREAD

**Discussion by EmilyMacias** [CS342/.github#48] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

Hello! I recently opened a pull request for my edits to the contact page of our team's app, but it looks like it fails the REUSE test because I uploaded an image for the contact picture and the new app icon. Here is the link to my pull request: https://github.com/CS342Industry/2024-Nourish/pull/2. What is the best way to fix this? Thank you!

### Reproduction

https://github.com/CS342Industry/2024-Nourish/pull/2

### Expected behavior

I expected it to pass all of the tests, but it looks like some copyright issue is happening based on the error message.

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by adritrao** [comment, marked as answer]
Hi @EmilyMacias,

To fix the copyright issue, you will need to add a .license file for the contact image, logo, and json file. You can create a license file by the name ```NourishLogo (1).png.license```,```anne.jpeg.license```, and ```Contents.json.license``` in the folders where you have added the images. You can copy over what is written in the other license files (e.g. [this file](https://github.com/CS342Industry/2024-Nourish/blob/part1/Nourish/Resources/Assets.xcassets/AppIcon.appiconset/AppIcon.png.license)) to the new ones. 

<img width="694" alt="Screenshot 2024-01-15 at 12 04 53 AM" src="https://github.com/CS342/.github/assets/67882813/08ab388d-2b69-4580-aaf4-faa14fd109bf">

This should fix the issue! Let me know if it doesn't work. 

Thanks!
Adrit

**Reply by EmilyMacias** [reply]
This worked! One follow up question is how you know what licensing files you should add and when. For example, both anne and NourishLogo refer to images but anne's image required two different licensing files. Why is that so? Thank you!

**Reply by vishnuravi** [reply]
The [REUSE Specification](https://reuse.software/spec/) requires that all files in your project have a license. This license could be inline in the file itself, if it is code (e.g., as a comment). If it is a binary file, such as an image, or any other type of file where you cannot add a license header inline, then we put in a separate file with the same name + the `.license` extension and place the license there.

By two files for anne's image, I think you are including the `Contents.json` file. This is just a file that contains metadata about the images stored in that directory - essentially a "Table of Contents". The NourishLogo is in another directory, which already has both a Contents.json and a corresponding Contents.json.license, which is why you did not need to make a license for the Contents.json there.
